### PR TITLE
Reading Big GeoTiffs

### PR DESCRIPTION
**Pull request by jbouffard**
This PR adds the ability for GeoTrellis to read Big, GeoTiff files. It is important to note that this **will only work with GeoTiffs**. Regular BigTiff files are currently not supported even though their smaller counterparts are. This is an issue that will be addressed and fixed in the near future.

### CODE REVIEW

**Comment by jbouffard** [comment]
I should also mention that I added three new files to the `geotiff-test-files.zip` [link](https://github.com/jbouffard/geotrellis/blob/4b316edd2229655d9def58c3850ee5bdabce6959/raster-test/data/geotiff-test-files.zip). Because of that, it is now *just* over the max file size of GitHub, though, it still uploaded fine. It's 50.8 MB while the max is 50 MB.

File: raster/src/main/scala/geotrellis/raster/io/geotiff/tags/codes/TiffFieldType.scala [hunk 1]
```diff
@@ -14,5 +14,7 @@ object TiffFieldType {
   val SignedFractionalsFieldType = 10
   val FloatsFieldType = 11
   val DoublesFieldType = 12
-
+  val LongsFieldType = 16
```
Comment (jbouffard) [inline raster/src/main/scala/geotrellis/raster/io/geotiff/tags/codes/TiffFieldType.scala]: This is 64 bits

Comment (jamesmcclain) [inline raster/src/main/scala/geotrellis/raster/io/geotiff/tags/codes/TiffFieldType.scala]: I would suggest something like

```scala
val LongsFieldType = 16 // This is 64-bits
```

Just to make sure it is super-obvious that this field is not present in normal TIFFs (that is, "long" means 64-bits in the Java world, whereas others might expect "long long").

**Comment by jbouffard** [comment]
@jamesmcclain Unfortuantely, even when I created a correct version of `all-ones.tif` and then made a BigTiff version of that with `gdal_translate` it still failed.

**Comment by jamesmcclain** [comment]
> @jamesmcclain Unfortuantely, even when I created a correct version of all-ones.tif and then made a BigTiff version of that with gdal_translate it still failed.

Ah, okay.  If we can just confirm that the actual bytes in the offset field of entry 34737 match those of the too-large value, that would be good enough for me.

**Comment by jbouffard** [comment]
@jamesmcclain I'm going to take this PR out of WIP, since it seems to work except for this one file; but I'll compare the bytes of the two GeoTiffs in a bit.

**Comment by jbouffard** [comment]
@lossyrob This is ready for review.

**Comment by jamesmcclain** [comment]
> @jamesmcclain I'm going to take this PR out of WIP, since it seems to work except for this one file; but I'll compare the bytes of the two GeoTiffs in a bit.

Okay, sounds like a winner.

File: raster/src/main/scala/geotrellis/raster/io/geotiff/reader/TiffTagsReader.scala [hunk 2]
```diff
@@ -29,19 +29,31 @@ object TiffTagsReader {
 
     // Validate GeoTiff identification number
     val geoTiffIdNumber = byteReader.getChar
-    if ( geoTiffIdNumber != 42)
-      throw new MalformedGeoTiffException(s"bad identification number (must be 42, was $geoTiffIdNumber)")
-
-    val tagsStartPosition = byteReader.getInt
-
-    read(byteReader, tagsStartPosition)
+    if ( geoTiffIdNumber != 42 && geoTiffIdNumber != 43)
```
Comment (lossyrob) [inline raster/src/main/scala/geotrellis/raster/io/geotiff/reader/TiffTagsReader.scala]: formatting: space after `if (`

**Comment by jbouffard** [comment]
@lossyrob Format change has been made.

**Pull request merged**